src/S3/S3ClientTrait.php

```diff
@@ -260,6 +261,30 @@ public function doesBucketExist($bucket)
         );
     }
 
+    /**
+     * @see S3ClientInterface::doesBucketExistV2()
+     */
+    public function doesBucketExistV2($bucket, $accept403 = false)
+    {
+        $command = $this->getCommand('HeadBucket', ['Bucket' => $bucket]);
+
+        try {
+            $this->execute($command);
+            return true;
+        } catch (S3Exception $e) {
```

Comment:
if it's a different kind of exception, it'll just get thrown back to the user, right?

Reply:
yeah, we're only interested in inspecting S3exceptions for true/false conditions and the rest can be thrown back to the user.